### DISCUSSION TITLE
state feedback controller design(based on reduction model)

### DISCUSSION THREAD

**Discussion by Leona-kk** [CURENT/andes#468] [Modeling] [answered]
Hi Dr. Cui and Jining,

Normally, we need model reduction and design a feedback controller for the larger system. Now, for Kundur system, I have reduced the model for 52*52 matrix to 2*2 (for example) and designed an LQR for this reduction system. But here the problem is how I can add a reduction model controller to the original system in the Andes and do TDS.

**Comment by jinningwang** [comment, marked as answer]
I feel it needs some extra design efforts.
1. In your reduction model, determine the controller's input and output variables. For example, the controller might take the variable ``omega`` of SynGen as input and ``tm`` of SynGen as output.
1. Developing a controller model. The model will take ``omega`` as input (reference to [self.omega = ExtState(...)](https://github.com/CURENT/andes/blob/master/andes/models/governor/tgbase.py#L100) for how to get the source variable), and add to ``pref`` (refer to [self.tm = ExtAlgeb(...)](https://github.com/CURENT/andes/blob/master/andes/models/governor/tgbase.py#L114) for how to feedback it to the target variable).

Welcome further discussion if you can give more details of your desired controllers.

**Reply by Leona-kk** [reply]
Hi jining, 
Thank you so much for the suggestions. I will consider this way.
